examples/GroupCharacters.jl

```diff
@@ -0,0 +1,350 @@
+degree(chi)
+length(chi)
+
+Main.GroupCharacters.scalar_product(t[1], t[1])
+Main.GroupCharacters.scalar_product(t[1], t[2])
+
+# a Brauer character table
+tmod2 = mod(t, 2);
+print(tmod2)
+show(tmod2)
+
+tmod2[2]  # a character
+tmod2[2,3]  # a character value
+
+length(tmod2) # number of conjugacy classes
+Main.GroupCharacters.trivial_character(tmod2)
+
+Main.GroupCharacters.decomposition_matrix(tmod2)
+
+# an ordinary character table computed from a group
+G = Oscar.symmetric_group(4)
+t = Main.GroupCharacters.character_table(G);
+print(t)
+show(t)
+
+t[2]
+t[2,3]
+
+# the corresponding Brauer table (possible because the group is solvable)
+tmod2 = mod(t, 2);
+print(tmod2)
+show(tmod2)
+
+# in general, GAP cannot compute Brauer tables from groups
+G = Oscar.symmetric_group(5)
+t = Main.GroupCharacters.character_table(G);
+mod(t, 2)
+```
+"""
+module GroupCharacters
+
+using Oscar
+
+# character values are elements from QabField
+#using QabModule
+#import QabModule: QabElem
+include("QabAndPChars.jl")
+
+import Base: getindex, length, mod, one, print, show, zero
+
+import Nemo: degree
+
+# Load GAP's character table library (if it is installed)
+GAP.Packages.load("ctbllib")
+
+
+#############################################################################
+##
+##  conversion between Julia's QabElem and GAP's cyclotomics
+##
+function QabModule.QabElem(cyc::Union{GAP.GapObj,Int64})
+    GAP.Globals.IsCyc(cyc) || error("cyc must be a GAP cyclotomic")
+    denom = GAP.Globals.DenominatorCyc(cyc)
+    n = GAP.Globals.Conductor(cyc)
+    coeffs = GAP.Globals.ExtRepOfObj(cyc * denom)
+    cycpol = GAP.Globals.CyclotomicPol(n)
+    GAP.Globals.ReduceCoeffs(coeffs, cycpol)
+    coeffs = GAP.gap_to_julia(coeffs, recursive = true)
+    coeffs = coeffs[1:(length(cycpol)-1)]
+    denom = GAP.gap_to_julia(denom)
+    F, z = Nemo.CyclotomicField(n)
+    val = Nemo.elem_from_mat_row(F, Nemo.matrix(Nemo.ZZ, [coeffs]), 1, fmpz(denom))
+    return QabModule.QabElem(val,n)
+end
+
+#function QabModule.QabElem(cyc::GAP.GapObj, c::Int)
+#end
+
+function gap_cyclotomic(elm::QabModule.QabElem)
+    coeffs = [Nemo.coeff(elm.data, i) for i in 0:(elm.c-1)]  # fmpq
+    coeffs = [GAP.julia_to_gap(BigInt(numerator(x)) // BigInt(denominator(x)))
+              for x in coeffs]
+    return GAP.Globals.CycList(GAP.julia_to_gap(coeffs))
+end
+
+
+#############################################################################
+##
+##  character tables
+##
+abstract type GroupCharacterTable end
+#abstract type GroupCharacterTableOrdinary <: GroupCharacterTable end
+#abstract type GroupCharacterTableBrauer <: GroupCharacterTable end
+
+struct GAPGroupCharacterTable <: GroupCharacterTable
+    GAPTable::GAP.GapObj  # the character table object
+    characteristic::Int
+end
+
+struct GAPGroupCharacterTableWithGroup <: GroupCharacterTable
+    GAPGroup::Oscar.GAPGroup    # the underlying group
+    GAPTable::GAP.GapObj  # the character table object
+    characteristic::Int
+end
```

Comment:
O.k., I will define just one type.

I would regard the possibility to set a group later on as asking for trouble.
For example, consider two character tables without stored groups, which belong to a group and a subgroup, respectively. If somebody later sets (independently) incompatible groups, suddenly some calls will fail that worked before setting the groups.
(The worst case will then be that the inconsistency is not evident enough for running into errors, and one gets wrong results.)
In GAP, we had run into this problem, and the operation `CharacterTableWithStoredGroup` was introduced (which creates a new table) instead of the possibility to simply set a group.